File: src/main/java/com/nerdearla/workshop/model/Payment.java
package com.nerdearla.workshop.model;

public class Payment {

    private final String paymentId;
    private final String authorizationId;

    public Payment(PaymentBuilder builder) {
        this.paymentId = builder.paymentId;
        this.authorizationId = builder.authorizationId;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public String getAuthorizationId() {
        return authorizationId;
    }

    public static class PaymentBuilder {

        private final String paymentId;
        private final String authorizationId;

        public PaymentBuilder(PaymentOperation operation, PaymentAuthorization authorization) {
            this.paymentId = operation.getPaymentId();
            this.authorizationId = authorization.getId();
        }

        public Payment build() {
            return new Payment(this);
        }
    }
}

File: src/main/java/com/nerdearla/workshop/model/Payment.kt
package com.nerdearla.workshop.model

data class Payment(
    val id: String,
    val authorizationId: String,
    val traceNumber: String,
    val qrId: String,
    val paymentMethodId: String,
    val amount: Double,
    val buyerId: String,
    val sellerId: String,
    val fraudValidationId: String
)
